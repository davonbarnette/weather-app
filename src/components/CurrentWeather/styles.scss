@import "../../styles/global";

.current-weather {
  @include flex-common(row, space-between, center);

  .details {
    @include flex-common(column, flex-end, flex-end);
    font-size:14px;
    & > time {

      color:$light-gray;
    }
    .city {
      font-weight:600;
    }
    .description {
      @include flex-common(row, flex-start, center);
      .icon {
        width:20px;
        height:20px;
        margin-right:9px;
      }
    }
  }
  .temperature{
    font-size: 55px;
    color:$dark-gray;
    margin-right:12px;
    font-weight:600;
    line-height:1em;
  }
}