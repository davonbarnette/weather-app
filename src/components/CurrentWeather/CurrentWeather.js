import React from 'react';
import Moment from 'react-moment';
import moment from 'moment';
import './styles.scss';
import {OpenWeatherHandler} from "../../utils/OpenWeatherHandler";

const CurrentWeather = (props) => {
    const {main, name, weather} = props.currentWeather;
    const {temp} = main;

    let weatherDetails = weather[0];

    return (
        <section className='current-weather'>
            <div className='temperature'>
                {temp.toFixed(0)}&#8457;
            </div>
            <div className='details'>
                <div className='city'>{name}</div>
                <Moment format={'dddd hh:mm A'}/>
                {weatherDetails &&
                <div className='description'>
                    <img alt="open-weather-icon" className='icon' src={OpenWeatherHandler.getIconUrl(weatherDetails.icon)}/>
                    {weatherDetails.description
                        .split(' ')
                        .map(t => t.charAt(0).toUpperCase() + t.slice(1))
                        .join(' ')}
                </div>}
            </div>
        </section>
    )
}

export default CurrentWeather;