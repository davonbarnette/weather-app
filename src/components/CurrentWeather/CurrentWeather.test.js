import React from 'react';
import { render } from '@testing-library/react';
import CurrentWeather from "./CurrentWeather";

Date.now = jest.fn(() => new Date("2020-05-13T12:33:37.000Z"));

describe('Forecast Component', () => {
    test('Snapshot', () => {
        let props = {
            currentWeather:{
                main: {temp: 80.51},
                name: 'test_city',
                weather:[
                    {icon: '01d', description: 'clear sky'}
                ]
            }
        }

        const {container} = render(<CurrentWeather {...props}/>)
        expect(container.firstChild).toMatchSnapshot();
    })
})