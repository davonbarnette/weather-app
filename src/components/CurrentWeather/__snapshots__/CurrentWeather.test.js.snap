// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Component Snapshot 1`] = `
<section
  class="current-weather"
>
  <div
    class="temperature"
  >
    81
    â„‰
  </div>
  <div
    class="details"
  >
    <div
      class="city"
    >
      test_city
    </div>
    <time
      datetime="1589373217000"
    >
      Wednesday 08:33 AM
    </time>
    <div
      class="description"
    >
      <img
        alt="open-weather-icon"
        class="icon"
        src="https://openweathermap.org/img/w/01d.png"
      />
      Clear Sky
    </div>
  </div>
</section>
`;
